src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -27372,5 +27368,554 @@ public ValueTuple(T1 item1)
+            Assert.Equal(0, item8Underlying.TupleElementIndex);
+            Assert.Same(item8Underlying, item8Underlying.TupleUnderlyingField);
+
+            var item9 = tuple.GetMember<TupleElementFieldSymbol>("Item9");
+            Assert.Equal("System.Int32 (System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32).Item9",
+                item9.ToTestDisplayString());
+            Assert.Equal(8, item9.TupleElementIndex);
+
+            var item9Underlying = item9.TupleUnderlyingField;
+            Assert.Equal("Item2", item9Underlying.Name);
+            Assert.IsType<SubstitutedFieldSymbol>(item9Underlying);
+            Assert.Equal(1, item9Underlying.TupleElementIndex);
+            Assert.Same(item9Underlying, item9Underlying.TupleUnderlyingField);
+        }
+
+        [Fact]
+        public void RealFieldsAreNotWrapped()
+        {
+            string source = @"
+public class C
+{
+    public (int, int) M() => throw null;
+    public (int Item1, int Item2) M2() => throw null;
+    public (int a, int b) M3() => throw null;
+}
+
+namespace System
+{
+    public struct ValueTuple<T1, T2>
+    {
+        public T1 Item1;
+        public T2 Item2;
+        public int field;
+
+        public ValueTuple(T1 item1, T2 item2) => throw null;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            CompileAndVerify(comp, symbolValidator: verifyModule, sourceSymbolValidator: verifyModule);
+
+            static void verifyModule(ModuleSymbol module)
+            {
+                var isSourceSymbol = module is SourceModuleSymbol;
+
+                var type = (NamedTypeSymbol)module.GlobalNamespace.GetMember<NamespaceSymbol>("System").GetMembers("ValueTuple").Single();
+                if (isSourceSymbol)
+                {
+                    Assert.Equal("SourceNamedTypeSymbol: (T1, T2)", print(type));
+                    AssertEx.SetEqual(new[] {
+                        "SourceMemberFieldSymbolFromDeclarator: field",
+                        "SourceMemberFieldSymbolFromDeclarator: Item1",
+                        "SourceMemberFieldSymbolFromDeclarator: Item2" }, printFields(type));
+                }
+                else
+                {
+                    Assert.Equal("PENamedTypeSymbolGeneric: (T1, T2)", print(type));
+                    AssertEx.SetEqual(new[] {
+                        "PEFieldSymbol: field",
+                        "PEFieldSymbol: Item1",
+                        "PEFieldSymbol: Item2" }, printFields(type));
+                }
+
+                var tuple1 = (NamedTypeSymbol)module.GlobalNamespace.GetMember<MethodSymbol>("C.M").ReturnType;
+                Assert.Equal("ConstructedNamedTypeSymbol: (System.Int32, System.Int32)", print(tuple1));
+                AssertEx.SetEqual(new[] {
+                    "SubstitutedFieldSymbol: field",
+                    "TupleElementFieldSymbol: Item1",
```

Comment:
>TupleElementFieldSymbol [](start = 21, length = 23)

Why do we create this wrapper? How ```IsDefinition``` and other interesting APIs work for this symbol?

Reply:
We need this wrapper to capture locations from tuple syntax.
`// Add the underlying/real field as an element (wrapping mainly to capture location). It should have the default name.`

I'll add a couple additional checks for `IsDefinition`, but I'm not sure what is the correct expectation.

---
In reply to: [426887075](https://github.com/dotnet/roslyn/pull/44231#discussion_r426887075) [](ancestors = 426887075)